recipes/aws-libfabric/all/conanfile.py

```diff
@@ -0,0 +1,113 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.35.0"
+
+class LibfabricConan(ConanFile):
+    name = "aws-libfabric"
+    description = "AWS Libfabric"
+    topics = ("fabric", "communication", "framework", "service")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/aws/libfabric"
+    license = "BSD-2-Clause", "GPL-2.0-or-later"
+    settings = "os", "arch", "compiler", "build_type"
+    _providers = ['gni', 'psm', 'psm2', 'sockets', 'rxm', 'tcp', 'udp', 'usnic', 'verbs', 'bgq', 'shm', 'efa', 'rxd', 'mrail', 'rstream', 'perf', 'hook_debug']
+    options = {
+        **{ p: "ANY" for p in _providers },
+        **{
+            "shared": [True, False],
+            "fPIC": [True, False],
+            "with_libnl": [True, False],
+            "with_bgq_progress": [None, "auto", "manual"],
+            "with_bgq_mr": [None, "basic", "scalable"]
+        }
+    }
+    default_options = {
+        **{ p: "auto" for p in _providers if p not in ('efa', 'verbs', 'rxd') },
+        **{
+            "shared": False,
+            "fPIC": True,
+            "efa": "yes",
+            "verbs": "no",
+            "rxd": "no",
+            "with_libnl": False,
+            "with_bgq_progress": None,
+            "with_bgq_mr": None
+        }
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    _autotools = None
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        if self.options.get_safe('with_libnl'):
+            self.requires('libnl/3.2.25')
+
+    def validate(self):
+        if self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("The libfabric package cannot be built on Visual Studio.")
+        for p in self._providers:
+            if self.options.get_safe(p) not in ["auto", "yes", "no", "dl"] and not os.path.isdir(str(self.options.get_safe(p))):
+                raise ConanInvalidConfiguration("Option {} can only be one of 'auto', 'yes', 'no', 'dl' or a directory path")
```

Comment:
With the latest suggestion, the option values are checked by limiting the number of allowed options (not using `ANY`).
As a rule, the recipesin cci do not allow external paths.
Allowing this would result in packages that are not transferable from system to system.
External dependencies should be conan packages themselves.
```suggestion
```